frame/staking/src/lib.rs

```diff
@@ -1835,3 +2309,56 @@ impl<T, Reporter, Offender, R, O> ReportOffence<Reporter, Offender, O>
 		}
 	}
 }
+
+#[allow(deprecated)]
+impl<T: Trait> frame_support::unsigned::ValidateUnsigned for Module<T> {
+	type Call = Call<T>;
+	fn validate_unsigned(call: &Self::Call) -> TransactionValidity {
+		if let Call::submit_election_solution_unsigned(
+			winners,
+			compact,
+			validator_index,
+			signature,
+		) = call {
+			if let Some(queued_elected) = Self::queued_elected() {
+				if queued_elected.compute == ElectionCompute::Authority {
+					debug::native::debug!(
+						target: "staking",
+						"rejecting unsigned transaction because we already have one from an authority"
+					);
+					return InvalidTransaction::Future.into();
+				}
+			}
+
+			// check signature
+			let payload = (winners, compact, validator_index);
+			let current_validators = T::SessionInterface::keys::<T::KeyType>();
+			let (_, validator_key) = current_validators.get(*validator_index as usize)
+				.ok_or(TransactionValidityError::Invalid(InvalidTransaction::Custom(0u8).into()))?;
+
+			// unwrap the key if it exists.
+			let validator_key = validator_key.as_ref()
+				.ok_or(TransactionValidityError::Invalid(InvalidTransaction::Custom(1u8).into()))?;
+
+			let signature_valid = payload.using_encoded(|encoded_payload| {
+				validator_key.verify(&encoded_payload, &signature)
+			});
+
+			if !signature_valid {
+				return InvalidTransaction::BadProof.into();
+			}
+
+			Ok(ValidTransaction {
+				priority: TransactionPriority::max_value(),
```

Comment:
I think this should take score of the solution into consideration. If we have multiple solutions (coming from different validators), it's better to start off with the best one.

Reply:
Treated. Now the priority is basically the provided slot stake. It should nicely fit into u64 in most cases and it is the most indication of how good this is.